pennylane/interfaces/tf.py

```diff
@@ -138,6 +138,7 @@ def _TFQNode(*input_, **input_kwargs):
         }
 
         # evaluate the QNode
+        qnode.set_trainable_args({i for i, val in enumerate(requires_grad) if val})
```

Comment:
I wonder if this can be combined into a single `for` loop along with `requires_grad`

Reply:
I had the same thought. The only other solution I can think of is NumPy, which will be overkill/slower for small lists than a list comprehension.

This also seems to be the most pythonic approach: https://stackoverflow.com/questions/6294179/how-to-find-all-occurrences-of-an-element-in-a-list

Reply:
Good catch. Something like
```python
qnode.set_trainable_args([
    idx for idx, i in enumerate(input_)
    if isinstance(i, (Variable, tf.Tensor)) and should_record_backprop([tf.convert_to_tensor(i)])
]
```

Reply:
Done!